src/node_worker.cc

```diff
@@ -132,7 +132,16 @@ class WorkerThreadData {
  public:
   explicit WorkerThreadData(Worker* w)
     : w_(w) {
-    CHECK_EQ(uv_loop_init(&loop_), 0);
+    int ret = uv_loop_init(&loop_);
+    if (ret != 0) {
+      char err_buf[128];
+      uv_err_name_r(ret, err_buf, sizeof(err_buf));
+      std::string error_str = SPrintF("%s %s",
+        "ERR_WORKER_INIT_FAILED:",
+        err_buf);
+      w->custom_error_ = error_str.c_str();
```

Comment:
Dangling pointer.

Reply:
@bnoordhuis , what is your guidance here? (a) if I duplicate a string using `strdup` , the caller will have to free it later. but the field `w->custom_error_` is largely designed to hold static strings, so the caller cannot figure out which one to delete based on what. (b) if I store the `std::string` as is into `w->custom_error_`, then I have to change its type and usage in many places.

Reply:
To be clear, my original suggestion *was* changing the type of `w->custom_error_`, i.e. b)

Reply:
@addaleax, thanks. Addressed your suggestion.